test/spec/functions-openrosa.spec.js

```diff
@@ -1035,23 +1038,17 @@ describe( 'Custom "OpenRosa" functions', () => {
             expect( test1 ).to.throw( /Failed to execute/ );
 
             // Add custom function
-            g.win.XPathJS.customXPathFunction.add( 'comment-status', {
-                fn( a ) {
-                    const curValue = a.toString();
-                    let status = '';
-
-                    try {
-                        status = JSON.parse( curValue ).status;
-                    } catch ( e ) {
-                        console.error( 'Could not parse JSON from', curValue );
-                    }
-
-                    return new g.win.XPathJS.customXPathFunction.type.StringType( status );
-                },
-                args: [ {
-                    t: 'string'
-                } ],
-                ret: 'string'
+            g.win.XPathJS.customXPathFunction.add( 'comment-status', function( a ) {
+              if(arguments.length !== 1) throw new g.win.Error('Invalid args');
+              const curValue = a.v[0]; // {t: 'arr', v: [{'status': 'good'}]}
+              let status = '';
+              try {
+                  status = JSON.parse( curValue ).status;
+              } catch ( e ) {
+                  console.error( 'Could not parse JSON from', curValue );
+              }
+
+              return new g.win.XPathJS.customXPathFunction.type.StringType( status );
```

Comment:
That looks good. Feel free to change more if e.g. `XPathJS.customXPathFunction.type.StringType` is an awkward solution in your library. Will you have some documentation on how to add custom functions in your library?